pkg/mdm/mdmtest/apple.go

```diff
@@ -386,6 +386,18 @@ func (c *TestAppleMDMClient) TokenUpdate() error {
 	return err
 }
 
+func (c *TestAppleMDMClient) DeclarativeManagement(endpoint string) error {
```

Comment:
I had this if it saves you some typing, can be added later as well:
```suggestion

// DeclarativeManagement sends a DeclarativeManagement checkin request to the server.
//
// The endpoint argument is used as the value for the `Endpoint` key in the request payload.
//
// For more details check https://developer.apple.com/documentation/devicemanagement/declarativemanagementrequest
func (c *TestAppleMDMClient) DeclarativeManagement(endpoint string) error {
```